packages/react-router/modules/Router.js

```diff
@@ -46,8 +46,9 @@ class Router extends React.Component {
     };
   }
 
-  componentWillMount() {
```

Comment:
Technically, you could change the children of `<Router />` when component is already rendered, causing it to re-render. If it's in constructor, you won't get the message then. Possibly `static getDerivedStateFromProps` again? As it's called every time component receives props.

Reply:
@grzegorzjudas That's a good point. I think we could follow with `getDerivedStateFromProps` for `invariant`, since it's called in `Mounting` and `Updating` phases.